### PR TITLE
[Fix]ComposerView input text repositioning while typing

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🔗 Issue Links

Resolves #https://github.com/GetStream/ios-issues-tracking/issues/627

### 🎯 Goal

Avoid unnecessary button visibility changes.

### 📝 Summary

Height calculation and scroll position for the textview are causing weird UI glitches

### 🛠 Implementation

Move calculations in one place (layoutSubviews) and allow every flow to go through it. Used https://raw.githubusercontent.com/KennethTsang/GrowingTextView/master/Sources/GrowingTextView/GrowingTextView.swift as guide. Due to this fix we can now safely deprecate the setTextViewHeight method which contained a workaround for one of the UI Issues. One thing to note is that the placeholder label **should not** be embedded in the view as its constraints were conflicting with the text size calculations.

### 🧪 Manual Testing Notes

- Run the app and try the following scenarios
- - Type many lines and then tap the shrink/unshrink button
- - - text position should remain intact
- - Type many lines and paste images while having the caret in top, middle and bottom positions
- - - Images should be pasted as expected and text position should be maintained in every case
- - Paste a very long text
- - - The caret should move at the bottom

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

**Review by martinmitrevski — APPROVED** [review]
LGTM! An approval from @testableapple would also be needed before merging.

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Review by testableapple — APPROVED** [review]
LGTM 🙌

**Comment by opfeffer** [comment]
Excited for this 🎉 Does this also work when `shrinkInputButton` and `commandsButton` are hidden?

**Comment by nuno-vieira** [comment]
> Excited for this 🎉 Does this also work when `shrinkInputButton` and `commandsButton` are hidden?

@opfeffer Yes 👍

File: Sources/StreamChatUI/CommonViews/InputTextView/InputTextView.swift [hunk 1]
```diff
@@ -36,6 +36,8 @@ open class InputTextView: UITextView, AppearanceProvider {
         }
     }
 
+    open var placeholderContainer: UIStackView = .init()
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/CommonViews/InputTextView/InputTextView.swift:39]: Can we add a description of the goal of this view? 🙏

Comment (martinmitrevski) [reply Sources/StreamChatUI/CommonViews/InputTextView/InputTextView.swift:39]: Important not to slow down the CI today just for this though.

**Pull request merged**